packages/cli-repl/src/cli-repl.ts

```diff
@@ -1254,4 +1257,96 @@ export class CliRepl implements MongoshIOProvider {
         version
     );
   }
+
+  private setupOIDCTokenDumpListener() {
+    function tryParseJWT(
+      token: string | null | undefined,
+      redact: 'redact' | 'include-secrets'
+    ): unknown {
+      if (!token) return token;
+      const jwtParts = token.split('.');
+      if (
+        jwtParts.length === 3 &&
+        jwtParts.every(
+          (part) =>
+            Buffer.from(part, 'base64url')
+              .toString('base64url')
+              .replace(/=+$/, '') === part
```

Comment:
Trying to understand what does this condition ensure? 🤔

Reply:
Yeah, added a comment here 🙂 This bit is trying to validate that the parts are base64url-encoded (required for JWTs) by decoding it from base64url and re-encoding it back, then checking whether that's equal to the original string